### ISSUE TITLE
Inspect non-haml files

### ISSUE THREAD

**Issue by akito19** [brigade/haml-lint#242]
The haml-lint has `VALID_EXTENSIONS` constant. I realize that it is a constant for analyzing only `.haml` files.
It is true that when I execute haml-lint to whole of an application or directories, haml-lint is analyzed only `.haml` files. However when I execute it to files including non-haml files directly, then haml-lint also analyzed non-haml files.

```
% ls app/views/users
_form.html.haml      edit.html.haml       index.json.jbuilder  show.html.haml
_user.json.jbuilder  index.html.haml      new.html.haml        show.json.jbuilder
% haml-lint app/views/users
warning: parser/current is loading parser/ruby24, which recognizes
warning: 2.4.0-compliant syntax, but you are running 2.4.1.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
app/views/users/_form.html.haml:1 [E] Syntax: unterminated string meets end of file

5 files inspected, 1 lint detected
% haml-lint app/views/users/*
warning: parser/current is loading parser/ruby24, which recognizes
warning: 2.4.0-compliant syntax, but you are running 2.4.1.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
app/views/users/_form.html.haml:1 [E] Syntax: unterminated string meets end of file
app/views/users/_user.json.jbuilder:2 [W] RuboCop: Layout/TrailingBlankLines: Final newline missing.
app/views/users/index.json.jbuilder:1 [W] RuboCop: Layout/TrailingBlankLines: Final newline missing.
app/views/users/show.json.jbuilder:1 [W] RuboCop: Layout/TrailingBlankLines: Final newline missing.

8 files inspected, 4 lints detected
```

The `.jbuilder` extension files are inspected because RuboCop defines it as [target files](https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/target_finder.rb#L12).
I think that haml-lint should inspect haml files but should not inspect non-haml files.

At least, It is not good that total of inspected files are different in case of `$ haml-lint app/views/users` and `$ haml-lint app/views/users/*` .

Following is my environment.

```
% haml-lint -V
warning: parser/current is loading parser/ruby24, which recognizes
warning: 2.4.0-compliant syntax, but you are running 2.4.1.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
haml-lint 0.25.1
haml 5.0.1
rubocop 0.49.0
ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-darwin16]
```

Thank you.

**Comment by sds** [comment]
Hey @akito19,

There is a significant difference between running:

```bash
haml-lint app/views/users
```

...and

```bash
haml-lint app/views/users/*
```

The former passes a directory to HAML-Lint, which then recursively searches for files matching `VALID_EXTENSIONS`, as you have pointed out.

The latter uses a glob which is expanded by your shell (e.g. bash/zsh) to resolve to a list of files as arguments. You can see the result by running `echo app/views/users/*`:

```
app/views/users/_form.html.haml app/views/users/_user.json.jbuilder app/views/users/index.json.jbuilder app/views/users/show.json.jbuilder ...etc...
```

When you explicitly pass a file path to HAML-Lint, it assumes you wanted to lint that file, regardless of extension. This behavior makes sense, because otherwise you would need to explicitly rename a file in order to lint it, which is silly for a tool to require.

If you don't want to lint your `.jbuilder` files, use a more specific glob pattern, e.g.

```bash
haml-lint app/views/users/*.haml
```

Hope that helps!
